Utilities/Mergers/src/MergerAlgorithm.cxx

```diff
@@ -0,0 +1,109 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file Merger.cxx
+/// \brief Implementation of O2 Mergers, v0.1
+///
+/// \author Piotr Konopka, piotr.jan.konopka@cern.ch
+
+#include "Mergers/MergerAlgorithm.h"
+
+#include "Mergers/MergeInterface.h"
+
+#include <TH1.h>
+#include <TH2.h>
+#include <TH3.h>
+#include <THn.h>
+#include <TTree.h>
+#include <THnSparse.h>
+#include <TObjArray.h>
+
+namespace o2::mergers::algorithm
+{
+
+void merge(TObject* const target, TObject* const other)
```

Comment:
Has the `other` object to be valid outside after the call? If not you could use move-like approach and avoid cloning.

Reply:
I think it should be valid after the call. In the case of FullHistoryMerger we might still need to reuse it if it is not updated with a newer version. Cloning should be rare anyway, only if we don't have already an object with a certain name.